tests/test_misc.py

```diff
@@ -801,6 +807,40 @@ def test_reorged_deposit_status(revault_network, bitcoind):
         )
         wait_for(lambda: len(w.rpc.listvaults(["spending"])["vaults"]) == 1)
 
+    # Now the same dance with a canceled vault
+
+    # Re-confirm the vault, get it active, then unvault and cancel it.
+    bitcoind.rpc.sendrawtransaction(deposit_tx)
+    bitcoind.generate_block(1, wait_for_mempool=2)
+    vault = revault_network.stk_wallets[0].rpc.listvaults(
+        # NB: 'spending' because we reuse a vault that was previously spent!
+        ["spending"]
+    )["vaults"][0]
+    revault_network.cancel_vault(vault)
+
+    # If the deposit is not unconfirmed, nothing changes
+    bitcoind.simple_reorg(vault["blockheight"], shift=2)
+    for w in revault_network.stk_wallets + revault_network.man_wallets:
+        w.wait_for_logs(
+            [
+                "Detected reorg",
+                f"Vault deposit '{deposit}' still has .* confirmations",
+                "Rescan of all vaults in db done.",
+            ]
```

Comment:
nit: You're saying "nothing changes" without testing that nothing has changed? Maybe check the vault status here?

Reply:
Right